presto-docs/src/main/sphinx/develop/secrets-manager.rst

```diff
@@ -0,0 +1,40 @@
+=====================
+Secrets Manager
+=====================
+
+Presto supports Custom Secrets Manager Plugins to retrieve secrets from secrets manager implementations of users' choice.
+By default, Hashicorp Vault is integrated as part of presto-vault module. More implementations can be plugged into the framework by extending
+the Presto SPI.
+
+
+Implementation
+--------------
+
+``SecretsManagerFactory`` is responsible for creating a
+``SecretsManager`` instance. It also defines a ``SecretsManager``
+name which is used by the administrator in a Presto configuration.
+
+``SecretsManager`` implementations have the responsibility to:
+
+* Read relevant configurations for connecting to the secrets manager service.
+* Fetch the secrets for all catalogs that require secrets manager.
+
+The implementation of ``SecretsManager`` and ``SecretsManagerFactory``
+must be wrapped as a plugin and installed on the Presto cluster.
+
+Configuration
+-------------
+
+After a plugin that implements ``SecretsManager`` and
+``SecretsManagerFactory`` has been installed on the coordinator, it is
+configured using an ``etc/secrets-manager.properties`` file.
+
+The ``secrets-manager.name`` property is used by Presto to find a registered
+``SecretsManagerFactory`` based on the name returned by
+``SecretsManagerFactory.getName()``. Remaining configuration needed by the custom implementation should be loaded from plugin code itself.
+
+Example configuration file:
```

Comment:
Is this example configuration file a complete example? 
Is anything else required in a ``secrets-manager.properties`` configuration file?
Is anything optional allowed to be in the `secrets-manager.properties`` configuration file?

Reply:
@steveburnett   could you please check the changes are satisfactory?